pages/shields/led_matrix.rst

```diff
@@ -1,4 +1,147 @@
 LED Matrix Shield
 *****************
 
-TODO
+.. highlight:: python
+   :linenothreshold: 4
+
+.. figure:: /images/led_matrix_shield_top_annotated.png
+   :width: 270
+
+   LED Matrix Shield, with row and bit ordering annotated
+
+The LED Matrix Shield is a Wemos D1 Mini shield, featuring an 8x8 LED Matrix
+of red LEDs.
+
+The shield has a TM1640 LED matrix driver on it that means we
+don't need to worry about how to make the matrix turn each LED on, we just
+send it a command with the LEDs we want enabled and it makes it happen!
+
+Further to this, there's a `MicroPython TM1640 driver`_ that has already been
+developed, and so we can use this to easily control the matrix in MicroPython.
+This ``tmp1640.py`` file from this should already be loaded onto your Wemos
+D1 Mini boards, so you should be set to get started!
+
+.. _MicroPython TM1640 driver: https://github.com/mcauser/micropython-tm1640
+
+Enter the Matrix
+================
+
+In order to start working with the LED matrix, we'll need to connect the shield
+to the Wemos D1 Mini board itself. If there is already a shield connected to
+your board (such as the RGB LED shield from the previous section), then first
+remove this. Then plug the LED Matrix shield into the board - the "LOLIN" label
+should be over the USB port of the main board. Then simply align the 8 pins on
+either side with the sockets on the main board and push them together!
+
+.. figure:: /images/led_matrix_shield_connected.png
+
+   Note the ``LOLIN`` text over the USB connection
+
+Now we've got the shield on the board, connect to the board with your USB
+again, and get into the REPL (by connecting to your device in your serial
+terminal software of choice). Now let's run the following commands to get
+these LEDs lit::
+
+    import machine
+    import wemos
+    import tm1640
+    tm = tm1640.TM1640(clk=machine.Pin(wemos.D5), dio=machine.Pin(wemos.D7))
+    tm.write([255,255,255,255,255,255,255,255])  # 255 = 0b11111111
```

Comment:
Maybe we should light this a little interesting so the orientation can be determined? Maybe even just make one row 0?

And we could introduce the binary format here instead of later? Seems pretty intuitive for this use...

Reply:
Something more interesting as an initial image might be a good idea, was more aiming for simplicity with initial lines (similarly with why I only show binary format later) - but may add confusion by aiming for simplicity. I'd be happy with this being changed as per your suggestions